.gitlab-ci.yml

```diff
@@ -46,53 +45,51 @@ variables:
     - cd $DOCKERFILE_DIR
     # - docker pull $CONTAINER_IMAGE:$CONTAINER_TAG || true
     # - docker build --cache-from $CONTAINER_IMAGE:$CONTAINER_TAG --tag $CONTAINER_IMAGE:$CI_BUILD_REF --tag $CONTAINER_IMAGE:$CONTAINER_TAG .
-    - |
-      if [ "${CONTAINER_DATE_TAG}" ]
-      then
-        docker build --no-cache --tag $CONTAINER_IMAGE:$CONTAINER_DATE_TAG --tag $CONTAINER_IMAGE:$CONTAINER_TAG .
-        docker push $CONTAINER_IMAGE:$CONTAINER_DATE_TAG
-      else
-        docker build --no-cache --tag $CONTAINER_IMAGE:$CONTAINER_TAG .
-      fi
+    - export CONTAINER_DATE_TAG="${CI_COMMIT_SHORT_SHA}-$(date +%Y%m%d)"
+    - docker build --no-cache
+        --build-arg VCS_REF="${CI_COMMIT_SHORT_SHA}"
+        --build-arg BUILD_DATE="$(date +%Y%m%d)"
+        --tag $CONTAINER_IMAGE:$CONTAINER_DATE_TAG
+        --tag $CONTAINER_IMAGE:$CONTAINER_TAG .
     - docker push $CONTAINER_IMAGE:$CONTAINER_TAG
   after_script:
     - docker logout
   environment:
     name: parity-build
 
 
-# Deprecated, use: parity/rust-parity-ethereum-build:xenial
-build:rust:nightly:
+# Deprecated, used only for substrate 1.0
+rust:
   <<: *docker_build
   variables:
-    CONTAINER_IMAGE: parity/rust
+    CONTAINER_IMAGE: parity/$CI_JOB_NAME
     CONTAINER_TAG:   nightly
     DOCKERFILE_DIR:  docker-files-for-Gitlab-CI-rust/rustup
   only:
     variables:
       - $DOCKERIMAGE == "parity/rust"
 
-build:rust-parity-ethereum-build:xenial:
+rust-parity-ethereum-build:
   <<: *docker_build
   variables:
-    CONTAINER_IMAGE: parity/rust-parity-ethereum-build
+    CONTAINER_IMAGE: parity/$CI_JOB_NAME
     CONTAINER_TAG:   xenial
-    DOCKERFILE_DIR:  docker-files-for-Gitlab-CI-rust/cross/android
+    DOCKERFILE_DIR:  docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci
```

Comment:
This is the target, I'll rename `docker-files-for-Gitlab-CI-rust` to be shorter(or just get rid of it?). Then exactly as you say: ` job name == image name == dockerfile dir`.
This will require changes everywhere: projects CIs, Dockerfiles, dirs. So will be done in the next iteration.